src/v/kafka/server/handlers/produce.cc

```diff
@@ -177,13 +165,13 @@ static partition_produce_stages partition_append(
   model::partition_id id,
   ss::lw_shared_ptr<replicated_partition> partition,
   model::batch_identity bid,
-  model::record_batch_reader reader,
+  model::record_batch batch,
   int16_t acks,
   int32_t num_records,
   int64_t num_bytes,
   std::chrono::milliseconds timeout_ms) {
     auto stages = partition->replicate(
-      bid, std::move(reader), acks_to_replicate_options(acks, timeout_ms));
+      bid, {std::move(batch)}, acks_to_replicate_options(acks, timeout_ms));
```

Comment:
AFAIK in initializer lists `std::move` doesn't eliminate the copy. Initializer list contains values and constructor copies from them.